### PR TITLE
JAX-WS and JAX-RS 2.1: Add Guard statements to CXF trace

### PR DESCRIPTION
**Pull request by neuwerk**
This PR adds guard statements to the recently added trace improvements in CXF.

### CODE REVIEW

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/attachment/AttachmentUtil.java [hunk 1]
```diff
@@ -113,6 +113,8 @@ public DataContentHandler createDataContentHandler(String mimeType,
 
         @Override
         public synchronized CommandInfo[] getAllCommands(String mimeType) {
+            
+            Boolean isFinestEnabled = LOG.isLoggable(Level.FINEST); // Liberty Change
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/attachment/AttachmentUtil.java]:
```suggestion
            boolean isFinestEnabled = LOG.isLoggable(Level.FINEST); // Liberty Change
```

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/jaxb/JAXBUtils.java [hunk 2]
```diff
@@ -1013,7 +1054,9 @@ private static void addToObjectFactoryCache(Package objectFactoryPkg,
             return;
         }
         synchronized (objectFactoryCache) {
-	    LOG.finest("Adding ObjectFactory package to cache: " + objectFactoryPkg); // Liberty Change
+            if (LOG.isLoggable(Level.FINEST)) { 
+                LOG.finest("Adding ObjectFactory package to cache: " + objectFactoryPkg); // Liberty Change
+            }
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/jaxb/JAXBUtils.java]: No reason for this to be inside the sync block.  Should just move it before the `synchronized (objectFactoryCache) {`

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java [hunk 3]
```diff
@@ -1952,9 +1980,13 @@ public static void printXmlFragment(XMLStreamReader reader) {
                 StaxUtils.close(writer);
             }
 			// Liberty Change - Prevent from going to message logs
-            LOG.finest(sw.toString());
+            if (LOG.isLoggable(Level.FINEST)) { // Liberty Change
+                LOG.finest(sw.toString());
+            }
         } catch (XMLStreamException e) {
-            LOG.severe(e.getMessage());
+            if (LOG.isLoggable(Level.FINEST)) { // Liberty Change
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: I don't think you want to have these check?  It is a change in behavior to no longer log a severe here unless trace is enabled.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyOutInterceptor.java [hunk 4]
```diff
@@ -134,7 +144,6 @@ protected void handle(Message msg) {
                 for (Assertion a : assertions) {
                     PolicyUtils.printPolicyComponent(a, buf, 1);
                 }
-                LOG.finest(buf.toString());
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyOutInterceptor.java:137]: Why did you remove this?  Now we don't even call LOG.finest?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/src/org/apache/cxf/wsdl/interceptors/DocLiteralInInterceptor.java [hunk 5]
```diff
@@ -133,7 +141,9 @@ public void handleMessage(Message message) {
                 }
 
             } else {
-		LOG.finest("This is Bare style");  // Liberty Change
+                if(LOG.isLoggable(Level.FINE)) { // Liberty Change
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/src/org/apache/cxf/wsdl/interceptors/DocLiteralInInterceptor.java]:
```suggestion
                if(LOG.isLoggable(Level.FINEST)) { // Liberty Change 
```
wrong check

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java (hunk 3 above)
Comment (neuwerk) [reply dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/staxutils/StaxUtils.java]: Yep, you're right. Must have been blindly adding it.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyOutInterceptor.java (hunk 4 above)
Comment (neuwerk) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyOutInterceptor.java:137]: It seemed redundant to me log the buffer immediately after calling `PolicyUtils.printPolicyComponent(...)`

Comment (neuwerk) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.policy.3.2/src/org/apache/cxf/ws/policy/PolicyOutInterceptor.java:137]: But on second glance that's not been added by Syed's changed, which I thought it was, so I will readd it.

**Review by jhanders34 — APPROVED** [review]

**Comment by neuwerk** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat, com.ibm.ws.jaxws.2.2.webcontainer_fat, com.ibm.ws.jaxws.2.2.webcontainer_fat_extended, com.ibm.ws.jaxws.cdi_fat, com.ibm.ws.jaxws.clientcontainer_fat, com.ibm.ws.jaxws.ejb_fat, io.openliberty.jaxrs.2.1.jaxws.2.2_fat, io.openliberty.jaxws.beanvalidation_fat,  io.openliberty.jaxws.global.handler.internal_fat, io.openliberty.jaxws.security_fat, io.openliberty.jaxws.security_fat.1, io.openliberty.jaxws.security_fat.ssl, io.openliberty.jaxws.tools_fat, io.openliberty.jaxws.wsat_fat, io.openliberty.jaxws_fat, com.ibm.ws.wsat.common_fat, com.ibm.ws.wsat.concurrent_fat, com.ibm.ws.wsat.migration_fat.1, com.ibm.ws.wsat.migration_fat.2, com.ibm.ws.wsat.migration_fat.3, com.ibm.ws.wsat.migration_fat.4, com.ibm.ws.wsat.migration_fat.5, com.ibm.ws.wsat.migration_fat.6, com.ibm.ws.wsat.migration_fat.7, com.ibm.ws.wsat.recovery_fat.lps, com.ibm.ws.wsat.recovery_fat.multi.1, com.ibm.ws.wsat.recovery_fat.multi.2, com.ibm.ws.wsat.recovery_fat.multi.3, com.ibm.ws.wsat.recovery_fat.multi.4, com.ibm.ws.wsat.recovery_fat.multi.5, com.ibm.ws.wsat.recovery_fat.multi.6, com.ibm.ws.wsat.recovery_fat.multi.7, com.ibm.ws.wsat.recovery_fat.single.1, com.ibm.ws.wsat.recovery_fat.single.2, com.ibm.ws.wsat_fat.assertion, com.ibm.ws.wsat_fat.db, com.ibm.ws.wsat_fat.db_optional, com.ibm.ws.wsat_fat.dbs, com.ibm.ws.wsat_fat.dbs_optional, com.ibm.ws.wsat_fat.multi, com.ibm.ws.wsat_fat.multi.misroute, com.ibm.ws.wsat_fat.ssl, com.ibm.ws.wsat_fat.ssl.misroute, io.openliberty.jaxws.wsat_fat, com.ibm.ws.wssecurity_fat.wsscxf.1, com.ibm.ws.wssecurity_fat.wsscxf.2, com.ibm.ws.wssecurity_fat.wsscxf.3, com.ibm.ws.wssecurity_fat.wsscxf.4, com.ibm.ws.wssecurity_fat.wsscxf.saml.1, com.ibm.ws.wssecurity_fat.wsscxf.saml.2, com.ibm.ws.wssecurity_fat.wsscxf.saml.3, com.ibm.ws.wssecurity_fat.wsscxf.saml.5, com.ibm.ws.wssecurity_fat.wsscxf.saml.6, io.openliberty.xmlws.4.0.internal_fat

**Comment by neuwerk** [comment]
#libby
